frontend/src/metabase/query_builder/selectors.js

```diff
@@ -616,20 +616,27 @@ export const getShouldShowUnsavedChangesWarning = createSelector(
     isResultsMetadataDirty,
     getQuestion,
     getIsSavedQuestionChanged,
+    getOriginalQuestion,
   ],
   (
     queryBuilderMode,
     isDirty,
     isMetadataDirty,
     question,
     isSavedQuestionChanged,
+    originalQuestion,
   ) => {
     const isEditingModel = queryBuilderMode === "dataset";
+    const isNewQuestion = !originalQuestion;
```

Comment:
Can we use `isSaved` here?

Reply:
No, `question.isSaved()` will be `false` after editing a question.

> `originalQuestion` is not reliable

I'm open to other suggestions.
BTW `getIsSavedQuestionChanged` also relies on `originalQuestion != null`.